pkgs/by-name/tg/tg-archive/package.nix

```
@@ -0,0 +1,47 @@
+{ lib
+, python3
+, fetchFromGitHub
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "tg-archive";
+  version = "1.1.3";
+  pyproject = true;
```

```suggestion
let
  pname = "tg-archive";
  version = "1.1.3";
  pyproject = true;
in
python3.pkgs.buildPythonApplication {
  inherit pname version pyproject;
```

Using `rec { ... }` is discouraged https://nix.dev/guides/best-practices#recursive-attribute-set-rec